logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/conversation/ConversationVerificationStatusHandlerTest.kt

```diff
@@ -0,0 +1,190 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.conversation.ConversationVerificationStatus
+import com.wire.kalium.logic.data.message.PersistMessageUseCase
+import com.wire.kalium.logic.framework.TestConversation
+import com.wire.kalium.logic.framework.TestUser
+import com.wire.kalium.logic.functional.Either
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.classOf
+import io.mockative.eq
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class ConversationVerificationStatusHandlerTest {
+
+    @Test
+    fun given_whenInvokingWithVerifiedStatus_thenInformedFlagSetFalse() = runTest {
+        val conversation = TestConversation.MLS_CONVERSATION
+        val (arrangement, handler) = Arrangement()
+            .arrange()
+
+        handler(conversation, ConversationVerificationStatus.VERIFIED)
+
+        verify(arrangement.conversationRepository)
+            .suspendFunction(arrangement.conversationRepository::setInformedAboutDegradedMLSVerificationFlag)
+            .with(eq(conversation.id), eq(false))
+            .wasInvoked(once)
+    }
+
+    @Test
+    fun given_whenInvokingWithNotVerifiedStatus_thenInformedFlagSetFalse() = runTest {
+        val conversation = TestConversation.MLS_CONVERSATION
+        val (arrangement, handler) = Arrangement()
+            .arrange()
+
+        handler(conversation, ConversationVerificationStatus.NOT_VERIFIED)
+
+        verify(arrangement.conversationRepository)
+            .suspendFunction(arrangement.conversationRepository::setInformedAboutDegradedMLSVerificationFlag)
+            .with(eq(conversation.id), eq(false))
+            .wasInvoked(once)
+    }
+
+    @Test
+    fun givenProteusConversation_whenInvokingWithDegradedStatus_thenInformedFlagSetFalse() = runTest {
+        val conversation = TestConversation.CONVERSATION
+        val (arrangement, handler) = Arrangement()
+            .arrange()
+
+        handler(conversation, ConversationVerificationStatus.DEGRADED)
+
+        verify(arrangement.conversationRepository)
+            .suspendFunction(arrangement.conversationRepository::setInformedAboutDegradedMLSVerificationFlag)
+            .with(eq(conversation.id), eq(false))
```

Comment:
I think here this should be 'true'?

Reply:
the setInformed here is false then, yes? I don't see the mock that was the missing logic that I wrote the above comment.

Reply:
for now we don't inform user about proteus degraded, as it's not implemented yet. So it's false, i'll add comment about it.